lib/chef/provider/service/macosx.rb

```diff
@@ -133,20 +149,32 @@ def disable_service
           unless @current_resource.enabled
             Chef::Log.debug("#{@new_resource} not enabled, not disabling")
           else
-            shell_out!(
-              "launchctl unload -w '#{@plist}'",
-              :user => @owner_uid, :group => @owner_gid
-            )
+            unload_service
           end
         end
 
+       def load_service
+          cmd = 'launchctl load -w '
+          cmd << "-S #{@session_type} " if @session_type
+          cmd << @plist
+          cmd = "#{@base_user_cmd} '#{cmd}'" if @console_user
+          shell_out_with_systems_locale!(cmd)
+        end
+
+        def unload_service
+          cmd = 'launchctl unload -w '
+          cmd << "-S #{@session_type} " if @session_type
+          cmd << @plist
+          cmd = "#{@base_user_cmd} '#{cmd}'" if @console_user
```

Comment:
I think you should use a helper method for running these commands. This would clear up what's going on with this string repeatedly, and reduce the duplication. Something like:

```
def shell_out_with_su(cmd)
  if @console_user
    shellout_with_systems_locale("#{@base_user_cmd} '#{cmd}'")
  else
    shellout_with_systems_locale(cmd)
  end
end
```